project-minimization-plugin/src/main/kotlin/org/plan/research/minimization/plugin/model/item/index/CompositeIndex.kt

```diff
@@ -0,0 +1,30 @@
+package org.plan.research.minimization.plugin.model.item.index
+
+import org.plan.research.minimization.plugin.psi.stub.KtStub
+
+import arrow.core.raise.option
+import com.intellij.psi.PsiElement
+import fleet.util.indexOfOrNull
+import org.jetbrains.kotlin.psi.KtClass
+import org.jetbrains.kotlin.psi.KtNamedFunction
+
+import kotlin.collections.filter
+
+sealed class CompositeIndex : PsiChildrenPathIndex {
```

Comment:
Why it's called composite? It seems that it doesn't "compose" any

Reply:
I thought about composing stubs and children's indexes. What do you think could be a better name?

Reply:
It's called PsiChildrenPathIndex :)

Reply:
Yes, I wanted to restrict the possible variants of it. Maybe I should not, shouldn't I?

Reply:
Considering your comments about the purpose of CompositeIndex, we can rename it, for example, like InstructionLookupIndex